src/components/badgeable/Badgeable.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+import './Badgeable.scss';
+
+type Props = {
+    /** Component to render when badging the bottom left corner of the rendered container */
+    bottomLeft?: React.ReactNode;
+    /** Component to render when badging the bottom right corner of the rendered container */
+    bottomRight?: React.ReactNode;
+    /** the item(s) to receive the badge */
+    children: React.ReactNode;
+    className?: string;
+    /** Component to render when badging the top left corner of the rendered container */
+    topLeft?: React.ReactNode;
+    /** Component to render when badging the top right corner of the rendered container */
+    topRight?: React.ReactNode;
+};
+
+const Badgeable = (props: Props) => {
+    const { children, className = '', topLeft = null, topRight = null, bottomLeft = null, bottomRight = null } = props;
+
+    return (
+        <div className={`badgeable-container ${className}`}>
```

Comment:
Should this component's CSS classes be converted to SUIT convention?

Should we use `classNames` here to avoid the extra space if `className` prop is empty?

Reply:
i would defer refactoring in this case, as there are likely many overrides associated with this component. That's why i moved to the mixins; we can then refactor to use those in consuming apps, and any class name changes automatically woke on future refactoring.